WordPress/Classes/Services/MediaService.m

```diff
@@ -230,6 +256,31 @@ - (void) createMediaForPost:(NSManagedObjectID *)postObjectID
     }];
 }
 
+- (void)configureNewMedia:(Media *)media
+             withMediaURL:(NSURL *)mediaURL
+        mediaThumbnailURL:(NSURL *)mediaThumbnailURL
+                mediaType:(MediaType)mediaType
+                mediaSize:(CGSize)mediaSize
+                    asset:(id <ExportableAsset>)asset
+{
+    media.filename = [mediaURL lastPathComponent];
+    media.absoluteLocalURL = mediaURL;
+    media.absoluteThumbnailLocalURL = mediaThumbnailURL;
+    NSNumber * fileSize;
+    if ([mediaURL getResourceValue:&fileSize forKey:NSURLFileSizeKey error:nil]) {
+        media.filesize = @([fileSize longLongValue] / 1024);
+    } else {
+        media.filesize = 0;
+    }
+    media.width = @(mediaSize.width);
+    media.height = @(mediaSize.height);
+    media.mediaType = mediaType;
+    if (mediaType == WPMediaTypeVideo && [asset isKindOfClass:[PHAsset class]]) {
```

Comment:
Okay, but I think I'd prefer to leave that code as-is for this PR – I just moved around what was there already.